File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int i=0;
        int l = nums.length;
        while(i < l){
            int correct = nums[i];
            if (correct < l && correct > 0 && nums[i] != nums[correct]){
                int temp = nums[i];
                nums[i] = nums[correct];
                nums[correct] = temp;
            }
            else{
                i++;
            }
        }
        for(int j=1;j<l;j++){
            if(nums[j] != j){
                return j;
            }
        }
        if(nums[0] > 0 && nums[0] == l){
            return l+1;
        }
        return l;
    }
}

File: 0041-first-missing-positive/0041-first-missing-positive.c
int firstMissingPositive(int* nums, int numsSize) {
    for(int i=0;i<numsSize;i++)
    {
        while(nums[i]<=numsSize && nums[i]>0 && nums[nums[i]-1]!=nums[i])
        {
            int temp=nums[nums[i]-1];
            nums[nums[i]-1]=nums[i];
            nums[i]=temp;

        }
    }
    for(int i=0;i<numsSize;i++)
    {
        if(nums[i]!=i+1)
        return i+1;
    }
    return numsSize+1;
}